### DISCUSSION TITLE
Webpack breaking modal dialog boxes in my portfolio site.

### DISCUSSION THREAD

**Discussion by jrobinson0529** [nss-evening-cohort-14/evening-client-side#108] [Help Me!] [answered]
**Jesse Robinson**

**Have you searched the other help tickets?** : Yes

I created a new repo with the template provided for Webpack. Copied and pasted in my scripts, styles, and index.html code similar to the sorting-hat-webpack challenge. I have project cards dynamically printed to the DOM with a button in the bottom right that used to toggle the modal to open and display an array of images that show each technology used for the projects. As of now the modal button will not trigger the modal to open.

**Only error is:**

GET http://localhost:8081/class=%22card-img-top%22 404 (Not Found)
Image (async) |   |  

**Google Searches**:

https://mdbootstrap.com/support/general/modal-not-working/
I don't really understand anything about jquery and it seems like people are seeking solutions using it.

https://stackoverflow.com/questions/17142790/bootstrap-modal-not-working-at-all/47235954
In this one they suggest adding a jquery function to a file in javascript but again, I do not understand the code line so I don't want to use it and I also don't know which file they are referencing to.

https://getbootstrap.com/docs/4.0/migration/
Found something about modals in the documentation but again I do not really understand what it means to run jQuery.load

The modal was previously part of the project card printer function. I have moved it to be hardcoded in the HTML. I changed the ID's on it temporarily to a static ID. I changed the ```data-bs-toggle="modal"``` and ```data-bs-target="modal"``` lines to data-toggle and data-target instead.

**main.scss**
https://github.com/jrobinson0529/portfolio/blob/jr-webpack/src/styles/main.scss 
**function to create project cards**
https://github.com/jrobinson0529/portfolio/blob/jr-webpack/src/javascripts/components/createProjectCards.js
**index.html**
https://github.com/jrobinson0529/portfolio/blob/7676a61fa7ee737e98f066b1e34fcf59d49f7e9f/src/index.html#L11-L26

**Comment by drteresavasquez** [comment]
@jrobinson0529 inside of your main.js file, try adding this line:

`import 'bootstrap';`

**Reply by drteresavasquez** [reply]
Link to resource: https://getbootstrap.com/docs/4.0/getting-started/webpack/

**Comment by TrinityChristiana** [comment]
[Line 24 of createProjectCards.js](https://github.com/jrobinson0529/portfolio/blob/590b90027ba23d2d674278888afd06fe65a9b37d/src/javascripts/components/createProjectCards.js#L24) is missing quotes around the image `src`. That might be your issue.

**Comment by jrobinson0529** [comment, marked as answer]
```import 'bootstrap';``` Worked I can't believe I ignored that when checking solutions I thought it was already covered by importing it through the scss file.. I still have issues with my bootstrap styles looking wonky. They aren't the way they were before and some of the bootstrap classes are just not working. I will check up on them again and update.

**Reply by jrobinson0529** [reply]
The modal is not smoothly opening and the font sizes have all changed. I have a lot of strange behavior on the page. For instance changing fs-3 to fs-5 in the modal body has no effect on the text size of the modal body. Same behavior on other elements as well.

**Reply by jrobinson0529** [reply]
Ok I think I am seeing the issue. I am using functionality from bootstrap 5.X which apparently doesn't let you style font sizes inline among other things. I am just going to redo the styling on the page using only bootstrap 4.6 features which is what we have installed through webpack dependencies I believe.

**Reply by drteresavasquez** [reply]
Yep. Use the webpack config and you should be GTG!
